pkg/indexer/indexer.go

```diff
@@ -43,6 +45,55 @@ func NewIndexer(dbParams database.Params, log *logger.Logger) (*Indexer, error)
 	}, nil
 }
 
+func addressesInOutput(output iotago.Output) []iotago.Address {
+	var foundAddresses []iotago.Address
+
+	// Check for addresses in features
+	features := output.FeatureSet()
+	if senderBlock := features.SenderFeature(); senderBlock != nil {
+		foundAddresses = append(foundAddresses, senderBlock.Address)
+	}
+
+	// Check for addresses in unlock conditions
+	conditions := output.UnlockConditionSet()
+	if addressUnlock := conditions.Address(); addressUnlock != nil {
+		foundAddresses = append(foundAddresses, addressUnlock.Address)
+	}
+	if storageDepositReturn := conditions.StorageDepositReturn(); storageDepositReturn != nil {
+		foundAddresses = append(foundAddresses, storageDepositReturn.ReturnAddress)
+	}
+	if expiration := conditions.Expiration(); expiration != nil {
+		foundAddresses = append(foundAddresses, expiration.ReturnAddress)
+	}
+	if accountUnlock := conditions.ImmutableAccount(); accountUnlock != nil {
+		foundAddresses = append(foundAddresses, accountUnlock.Address)
+	}
+
+	// Check for addresses in immutable features
+	if chainOutput, ok := output.(iotago.ChainOutputImmutable); ok {
+		immutableFeatures := chainOutput.ImmutableFeatureSet()
+
+		if issuerBlock := immutableFeatures.Issuer(); issuerBlock != nil {
+			foundAddresses = append(foundAddresses, issuerBlock.Address)
+		}
+
+		if stateController := conditions.StateControllerAddress(); stateController != nil {
```

Comment:
why is the state conditions stuff like StateController and Governour under the ChainOutputImmutable?

Reply:
not strictly necessary, but this allows to skip those checks for basic outputs. I can move them out if the code is cleaner